MedtrumKitUI/ViewModels/Onboarding/PumpBaseSettingsViewModel.swift

```diff
@@ -21,10 +21,13 @@ class PumpBaseSettingsViewModel: ObservableObject {
         }
         
         self.serialNumber = pumpManager.state.pumpSN.hexEncodedString().uppercased()
-        self.is300u = pumpManager.state.pumpName.contains("300U")
+        if pumpManager.state.pumpSN.count == 4 {
```

Comment:
Maybe use a more thorough check here, in AAPS we use this:
https://github.com/nightscout/AndroidAPS/blob/ba551caffc9d0d0aaae5b1c5a98312417de04d02/pump/medtrum/src/main/kotlin/app/aaps/pump/medtrum/util/MedtrumSnUtil.kt#L16